src/groupmesh.cpp

```diff
@@ -236,18 +237,14 @@ void Group::GenerateShellAndMesh() {
             // And for any plane faces, annotate the model with the entity for
             // that face, so that the user can select them with the mouse.
             Vector onOrig = sbls->point;
-            int i;
-            // Not using range-for here because we're starting at a different place and using
-            // indices for meaning.
-            for(i = is; i < thisShell.surface.n; i++) {
-                SSurface *ss = &(thisShell.surface[i]);
+            for(auto ss : indices(thisShell.surface, is)) {
                 hEntity face = Entity::NO_ENTITY;
 
                 Vector p = ss->PointAt(0, 0),
                        n = ss->NormalAt(0, 0).WithMagnitude(1);
                 double d = n.Dot(p);
 
-                if(i == is || i == (is + 1)) {
+                if(ss.GetIndex() <= (is + 1)) {
```

Comment:
This is not exactly the same code. But it does the same, because the loop starts from `is`.
The old code makes it a bit more obvious that you want the first two elements.

Reply:
yeah, I wasn't sure whether I should just make an `i` variable to leave this code unmodified, replace both uses of `i` with `ss.GetIndex()`, or do this. Replacing both looked long and unwieldy so I did this, but any of those would work.